### PR TITLE
Upgrade MacOS 10.15 to 12.5

### PR DESCRIPTION
**Pull request by jrw972**

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -11662,11 +11662,11 @@ jobs:
   #       cat "$GITHUB_WORKSPACE/OpenDDS/${{ github.job }}_autobuild_workspace/latest.txt"
   #       grep -q 'Failures: 0' "$GITHUB_WORKSPACE/OpenDDS/${{ github.job }}_autobuild_workspace/latest.txt"
 
-  # macOS 10.15
+  # macOS 12.5
```
Comment (mitza-oci) [inline .github/workflows/build_and_test.yml:11665]: Update github_actions.rst

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -11897,7 +11897,7 @@ jobs:
         <command name="check_compiler" options="gcc"/>
         <command name="print_perl_version"/>
         <command name="print_autobuild_config"/>
-        <command name="auto_run_tests" options="script_path=$ACE_ROOT/bin dir=$GITHUB_WORKSPACE/OpenDDS -l tests/core_ci_tests.lst -Config STATIC_MESSENGER -Config STATIC_COMPILER -Config STATIC_COMPILER2 -Config STATIC_UNIT_TESTS -Config XERCES3 -Config OPENDDS_SECURITY -Config CXX11 -Config RAPIDJSON -Config NO_SHMEM -Config DDS_NO_ORBSVCS -Config GH_ACTIONS_M10"/>
+        <command name="auto_run_tests" options="script_path=$ACE_ROOT/bin dir=$GITHUB_WORKSPACE/OpenDDS -l tests/core_ci_tests.lst -Config STATIC_MESSENGER -Config STATIC_COMPILER -Config STATIC_COMPILER2 -Config STATIC_UNIT_TESTS -Config XERCES3 -Config OPENDDS_SECURITY -Config CXX11 -Config RAPIDJSON -Config NO_SHMEM -Config DDS_NO_ORBSVCS -Config GH_ACTIONS_M12"/>
```
Comment (sonndinh) [inline .github/workflows/build_and_test.yml:11900]: We probably need to update the test list files such as `dcps_tests.lst` to replace `GH_ACTIONS_M10` with this new flag for the tests where it applies.

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -11792,6 +11792,8 @@ jobs:
         make -j4
     - name: Build CMake Tests
       run: |
+        echo Show setenv.sh
+        cat OpenDDS/setenv.sh
```
Comment (sonndinh) [inline .github/workflows/build_and_test.yml:11796]: These lines were just for diagnosing. We can remove them since the build is successful now.

**Review by mitza-oci — APPROVED** [review]

File: docs/internal/github_actions.rst [hunk 4]
```diff
@@ -18,7 +18,7 @@ Operating System
 
 * u18/u20/u22 - Ubuntu 18/20/22.04
 * w19/w22 - Windows Server 2019/Windows Server 2022
-* m10/m11 - macOS 10.15/11
+* m10/m11/m12 - macOS 10.15/11/12
```
Comment (sonndinh) [inline docs/internal/github_actions.rst:21]:
```suggestion
* m11/m12 - macOS 11/12
```

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (jrw972) [reply .github/workflows/build_and_test.yml:11900]: The tests are passing so, I think I will leave it alone.

**Pull request merged**